ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala

```diff
@@ -89,9 +91,10 @@ trait WriteService {
     *   transaction id.
     *
     */
+  //TODO: update docs
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
-      transaction: SubmittedTransaction): Unit
+      transaction: SubmittedTransaction): Future[SubmissionResult]
```

Comment:
as discussed, lets use java-friendly types at this level, e.g. CompletableFuture.

Reply:
changed it to use the interface `CompletionStage`